packages/logger/src/lib/logger-class-support.ts

```diff
@@ -0,0 +1,78 @@
+import { createLogger } from './create-logger';
+import { LogConsumer } from './model/log-consumer';
+import { LogLevel } from './model/log-level';
+
+export interface ClassLogger {
+  logLevel: LogLevel;
+
+  debug(source: unknown, ...messages: unknown[]): void;
+
+  info(source: unknown, ...messages: unknown[]): void;
+
+  warn(source: unknown, ...messages: unknown[]): void;
+
+  error(source: unknown, ...messages: unknown[]): void;
+
+  addConsumer(consumer: LogConsumer): void;
+}
+
+const classLoggerMarker = Symbol('ClassLoggerMarker');
+
+export function LoggerMark(name: string) {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return (target: any): any => {
+    Object.defineProperty(target.prototype, classLoggerMarker, {
+      configurable: false,
+      enumerable: false,
+      writable: false,
+      value: name,
+    });
+
+    return target;
+  };
+}
+
+export function createClassLogger(logger = createLogger()): ClassLogger {
```

Comment:
we should rather use the global logger instance, otherwise there will not be any consumers registered